src/hashtable.py

```diff
@@ -67,14 +68,15 @@ def remove(self, key):
 
 
     def retrieve(self, key):
-        '''
-        Retrieve the value stored with the given key.
+        index = self._hash_djb2(key)
 
-        Returns None if the key is not found.
+        if self.storage[index] is None:
+            return None
+
+        value = self.storage[index].value
```

Comment:
you're chaining with your insert but not retrieve? don't know which tests aren't passing but it's something to fix.